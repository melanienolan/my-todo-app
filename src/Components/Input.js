import React from 'react';
import PropTypes from 'prop-types';
import Plus from 'react-icons/lib/fa/plus';

const Input = props => {
  return (
    <div className="add-todo">
      <input
        className="add-todo__input"
        type="text"
        value={props.inputValue}
        onChange={event => props.onUpdate(event.target.value)}
        onKeyUp={event =>
          (event.keyCode === 13 ? props.onAddTodoSubmit() : false)}
      />
      <button
        className="add-todo__button"
        onClick={() => props.onAddTodoSubmit()}
      >
        <Plus />
      </button>
    </div>
  );
};

Input.propTypes = {
  inputValue: PropTypes.string,
  onAddTodoSubmit: PropTypes.func,
  onUpdate: PropTypes.func
};

export default Input;
