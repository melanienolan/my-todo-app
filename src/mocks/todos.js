export const todos = [
  {
    completed: true,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'complete todo'
  },
  {
    completed: false,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'incomplete todo'
  }
];
