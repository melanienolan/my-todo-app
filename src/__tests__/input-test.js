import React from 'react';
import Input from '../Components/Input';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { inputValue } from '../mocks/todos';

it('renders input component without crashing', () => {
  shallow(<Input inputValue={inputValue} />);
});

it('renders todo component with todo items', () => {
  const inputComponent = shallow(<Input inputValue={inputValue} />);
  expect(toJson(inputComponent)).toMatchSnapshot();
});
