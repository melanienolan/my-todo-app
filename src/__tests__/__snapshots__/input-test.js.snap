// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input component with todo items 1`] = `
<div
  className="add-todo"
>
  <input
    className="add-todo__input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Add todo..."
    type="text"
    value="Here is the input value"
  />
  <button
    className="add-todo__button"
    onClick={[Function]}
  >
    <FaPlus />
  </button>
</div>
`;
